vespalib/src/vespa/vespalib/btree/btreeiterator.h

```diff
@@ -451,6 +483,54 @@ class BTreeIteratorBase
             _leafRoot->foreach_key(func);
         }
     }
+
+    template <typename FunctionType>
+    void
+    foreach_key_range(const BTreeIteratorBase &end_itr, FunctionType func) const
+    {
+        if (!valid()) {
+            return;
+        }
+        if (!end_itr.valid()) {
+            foreach_key_range_start(_pathSize, func);
+            return;
+        }
+        assert(_pathSize == end_itr._pathSize);
+        assert(_allocator == end_itr._allocator);
+        uint32_t level = _pathSize;
+        if (level > 0u) {
+            uint32_t idx;
+            uint32_t eidx;
+            do {
+                --level;
+                assert(_path[level].getNode() == end_itr._path[level].getNode());
+                idx = _path[level].getIdx();
+                eidx = end_itr._path[level].getIdx();
+                if (idx > eidx) {
+                    return;
+                }
+                if (idx != eidx) {
+                    ++level;
+                    break;
+                }
+            } while (level != 0);
+            if (level > 0u) {
+                foreach_key_range_start(level - 1, func);
```

Comment:
Please add comment describing the 3 steps done here (left, middle, and right sub trees).

Reply:
Done.